pp_pack.c

```diff
@@ -842,8 +842,11 @@ Perl_unpackstring(pTHX_ const char *pat, const char *patend, const char *s, cons
         /* We probably should try to avoid this in case a scalar context call
            wouldn't get to the "U0" */
         STRLEN len = strend - s;
-        s = (char *) bytes_to_utf8((U8 *) s, &len);
-        SAVEFREEPV(s);
+        void * free_me = NULL;
+        s = (char *) bytes_to_utf8_free_me((U8 *) s, &len, &free_me);
+        if (free_me) {
+            SAVEFREEPV(s);
```

Comment:
Technically this should `SAVEFREEPV(free_me);`

I do wonder if there should be a bytes_to_utf8_temp_pv().

Reply:
`bytes_to_utf8_temp_pv` added and this changed to use that.